### PR TITLE
Added quick start template for creating a budget

### PR DESCRIPTION
**Pull request by asarkar84**
Added quick start template for creating a budget

### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: create-budget/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Create Budget to track cost or usage
```
Comment (bmoore-msft) [inline create-budget/README.md]: Use title casing

File: create-budget/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F2create-budget%2Fazuredeploy.json" target="_blank">
+    <img src="http://azuredeploy.net/deploybutton.png"/>
+</a>
+<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fcreate-budget%2Fazuredeploy.json" target="_blank">
+    <img src="http://armviz.io/visualizebutton.png"/>
+</a>
+
+This template shows how to create a budget to track cost/usage and get notified whenever a specified threshold is met.
+
+The accompanying PowerShell script shows how to create a resource group and budget under that resource group from the template. Before running the script, edit *azuredeploy.parameters.json* and replace the values marked with *'#####'* and *'YYYY-MM-DD'* and other sample data.
```
Comment (bmoore-msft) [inline create-budget/README.md]: update this:
- there's no script
- the parameter updates need to be more detailed given the failures we had

File: create-budget/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,160 @@
+﻿{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "budgetName": {
+      "type": "string",
+      "metadata": {
+        "description": "Budget name"
```
Comment (bmoore-msft) [inline create-budget/azuredeploy.json]: What are the naming reqs for this?  i.e. is it globally unique?  State one way or another - add a defaultValue as appropriate

File: create-budget/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+        "Quarterly",
+        "Annually"
+      ],
+      "metadata": {
+        "description": "The time covered by a budget. Tracking of the amount will be reset based on the time grain."
+      }
+    },
+    "startDate": {
+      "type": "string",
+      "defaultValue": "YYYY-MM-DD",
```
Comment (bmoore-msft) [inline create-budget/azuredeploy.json]: remove this since it's not a valid defaultValue

File: create-budget/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+      "metadata": {
+        "description": "The start date for the budget."
+      }
+    },
+    "endDate": {
+      "type": "string",
+      "defaultValue": "YYYY-MM-DD",
```
Comment (bmoore-msft) [inline create-budget/azuredeploy.json]: remove this

File: create-budget/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+    },
+    "threshold": {
+      "type": "string",
+      "metadata": {
+        "description": "Threshold value associated with a notification. Notification is sent when the cost exceeded the threshold. It is always percent and has to be between 0 and 1000."
+      }
+    },
+    "contactEmails": {
+      "type": "array",
+      "defaultValue": [
```
Comment (bmoore-msft) [inline create-budget/azuredeploy.json]: remove the defaultValue

File: create-budget/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "budgetName": {
+      "type": "string",
+      "metadata": {
+        "description": "Budget name"
+      }
+    },
+    "amount": {
+      "type": "string",
```
Comment (bmoore-msft) [inline create-budget/azuredeploy.json]: add a defaultValue

File: create-budget/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+        "EqualTo",
+        "GreaterThan",
+        "GreaterThanOrEqualTo"
+      ],
+      "metadata": {
+        "description": "The comparison operator."
+      }
+    },
+    "threshold": {
+      "type": "string",
```
Comment (bmoore-msft) [inline create-budget/azuredeploy.json]: add a defaultValue

File: create-budget/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+        "Contributor",
+        "Reader"
+      ],
+      "metadata": {
+        "description": "Contact roles to send the budget notification to when the threshold is exceeded."
+      }
+    },
+    "contactGroups": {
+      "type": "array",
+      "defaultValue": [
```
Comment (bmoore-msft) [inline create-budget/azuredeploy.json]: remove this

File: create-budget/azuredeploy.json [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+        "/subscriptions/{subscription-id}/resourceGroups/{resource-group-name}/providers/microsoft.insights/actionGroups/SampleActionGroup1"
+      ],
+      "metadata": {
+        "description": "Contact roles to send the budget notification to when the threshold is exceeded."
+      }
+    },
+    "resourcesFilter": {
+      "type": "array",
+      "defaultValue": [
```
Comment (bmoore-msft) [inline create-budget/azuredeploy.json]: remove this

File: create-budget/azuredeploy.json [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+        "/subscriptions/{subscription-id}/resourceGroups/{resource-group-name}/providers/Microsoft.Compute/virtualMachines/vm1",
+        "/subscriptions/{subscription-id}/resourceGroups/{resource-group-name}/providers/Microsoft.Compute/virtualMachines/vm2"
+      ],
+      "metadata": {
+        "description": "The list of filters on resources."
+      }
+    },
+    "metersFilter": {
+      "type": "array",
+      "defaultValue": [
```
Comment (bmoore-msft) [inline create-budget/azuredeploy.json]: remove this

File: create-budget/azuredeploy.parameters.json [hunk 12]
```diff
@@ -0,0 +1,57 @@
+﻿{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "budgetName": {
+      "value": "#####"
```
Comment (bmoore-msft) [inline create-budget/azuredeploy.parameters.json]: add valid values here where possible - if they are subscription specific then you can use placeholders (e.g. resourceId)

File: create-budget/azuredeploy.json (hunk 8 above)
Comment (asarkar84) [reply create-budget/azuredeploy.json]: Added 90 as default.

File: create-budget/azuredeploy.json (hunk 7 above)
Comment (asarkar84) [reply create-budget/azuredeploy.json]: Added 1000 as default

File: create-budget/azuredeploy.json (hunk 6 above)
Comment (asarkar84) [reply create-budget/azuredeploy.json]: Sure! Removed.

File: create-budget/azuredeploy.json (hunk 5 above)
Comment (asarkar84) [reply create-budget/azuredeploy.json]: Sure! Removed.

File: create-budget/azuredeploy.json (hunk 4 above)
Comment (asarkar84) [reply create-budget/azuredeploy.json]: Sure! Removed.

File: create-budget/azuredeploy.json (hunk 3 above)
Comment (asarkar84) [reply create-budget/azuredeploy.json]: Updated the description. Budget name should be unique within the resource group. Adding MyBudget as Default value.

File: create-budget/README.md (hunk 1 above)
Comment (asarkar84) [reply create-budget/README.md]: Sure! Updated.

File: create-budget/azuredeploy.json (hunk 9 above)
Comment (asarkar84) [reply create-budget/azuredeploy.json]: Sure! Removed the default value here.

File: create-budget/azuredeploy.json (hunk 10 above)
Comment (asarkar84) [reply create-budget/azuredeploy.json]: Sure! I have removed the default value here.

File: create-budget/azuredeploy.json (hunk 11 above)
Comment (asarkar84) [reply create-budget/azuredeploy.json]: I have removed the default value here.

File: create-budget/azuredeploy.parameters.json (hunk 12 above)
Comment (asarkar84) [reply create-budget/azuredeploy.parameters.json]: Added valid values wherever feasible and removed invalid values. Added more details about each parameter in README.md file.

File: create-budget/README.md (hunk 2 above)
Comment (asarkar84) [reply create-budget/README.md]: Removed the mention about script. Also added detailed information regarding parameters.

**Comment by asarkar84** [comment]
Review comments have been incorporated. Please take a look.

**Pull request merged**